Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/PhysicsPicking.cpp

```diff
@@ -0,0 +1,189 @@
+                            const AZ::Vector3 capsuleZ = colliderGlobalTransform.GetBasisZ();
+                            const float cylinderHeight = AZ::GetMax(AZ::Constants::Tolerance, capsule->m_height - 2.0f * capsule->m_radius);
+                            const AZ::Vector3 cylinderEnd1 = colliderGlobalTransform.GetTranslation() - 0.5f * cylinderHeight * capsuleZ;
+                            const AZ::Vector3 cylinderEnd2 = colliderGlobalTransform.GetTranslation() + 0.5f * cylinderHeight * capsuleZ;
+                            const float rayLength = 1000.0f;
+                            float t = AZ::Constants::FloatMax;
+                            const AZ::Intersect::CapsuleIsectTypes result = AZ::Intersect::IntersectSegmentCapsule(
+                                rayOrigin, rayDirection * rayLength, cylinderEnd1, cylinderEnd2, capsule->m_radius, t);
+                            if (result != AZ::Intersect::CapsuleIsectTypes::ISECT_RAY_CAPSULE_NONE)
+                            {
+                                const float distance = rayLength * t;
+                                CompareIntersection(closestIntersection, distance, jointIndex);
+                            }
+                        }
+                        else if (colliderType == azrtti_typeid<Physics::BoxShapeConfiguration>())
+                        {
+                            auto* box = static_cast<Physics::BoxShapeConfiguration*>(shapeColliderPair.second.get());
+                            float distance = AZ::Constants::FloatMax;
+                            if (AZ::Intersect::IntersectRayBox(
+                                    rayOrigin,
+                                    rayDirection,
+                                    colliderGlobalTransform.GetTranslation(),
+                                    colliderGlobalTransform.GetBasisX(),
+                                    colliderGlobalTransform.GetBasisY(),
+                                    colliderGlobalTransform.GetBasisZ(),
+                                    0.5f * box->m_dimensions.GetX(),
+                                    0.5f * box->m_dimensions.GetY(),
+                                    0.5f * box->m_dimensions.GetZ(),
+                                    distance))
+                            {
+                                CompareIntersection(closestIntersection, distance, jointIndex);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        if (closestIntersection.m_intersected)
+        {
+            QModelIndex modelIndex = skeletonModel->GetModelIndex(skeleton->GetNode(closestIntersection.m_jointIndex));
+            skeletonModel->GetSelectionModel().select(modelIndex, QItemSelectionModel::ClearAndSelect | QItemSelectionModel::Rows);
+        }
+
+        return closestIntersection.m_intersected;
+    }
+
+    void PhysicsPicking::UpdateRenderFlags(ActorRenderFlags renderFlags)
```

Comment:
nit: would it fit better to call it `SetRenderFlags`?

Reply:
Yeah, good idea, done.